hal/src/nRF52840/ble_hal.cpp

```diff
@@ -2366,13 +2332,6 @@ int BleObject::ConnectionsManager::processConnectedEventFromThread(const ble_evt
         linkEvent.params.connected.info = &connection.info;
         notifyLinkEvent(linkEvent);
     }
-    if (desiredAttMtu_ > BLE_DEFAULT_ATT_MTU_SIZE) {
```

Comment:
@XuGuohui This seems like a major behavior change where we now require the caller to use `hal_ble_gatt_client_att_mtu_exchange()`. Can we keep this automatic?

Reply:
What about keeping the existing automatic ATT MTU exchanging scheme, plus probably adding retries? And if it kept failing, this API still enable user to perform the exchange.

Reply:
Yeah, I think that'll work best. You get good out of the box behavior (we attempt to negotiate max possible MTU) and the app can always trigger it manually if required.